mono/metadata/boehm-gc.c

```diff
@@ -668,6 +668,8 @@ mono_gc_alloc_obj (MonoVTable *vtable, size_t size)
 	if (G_UNLIKELY (mono_profiler_events & MONO_PROFILE_ALLOCATIONS))
 		mono_profiler_allocation (obj);
 
+    mono_stats.new_object_count++;
```

Comment:
None of the other mono_stats are collected atomically, nor do they appear to have an #if enable wrapping them